pygfx/renderers/wgpu/_wgpurenderer.py

```diff
@@ -849,3 +962,116 @@ def _update_sampler(self, resource):
+        pos is a 2D point in logical pixels (with the origin at the
+        top-left). Returns a dict with fields:
+
+        * "rgba": The value in the color buffer. All zero's when rendering
+          directly to the screen (bypassing post-processing).
+        * "depth_value": The raw value in the depth buffer (0..1). A
+          depth value of 1.0 means that the background is picked.
+        * "position": The position in world coordinates.
+        * "distance_to_camera": The distance (in world units) from the position
+          to the camera position.
+        """
+
+        # Make pos 0..1, so we can scale it to the render texture
+        if self._logical_size:
+            logical_size = self._logical_size
+        else:
+            logical_size = self._canvas.get_logical_size()
+        float_pos = pos[0] / logical_size[0], pos[1] / logical_size[1]
+
+        can_sample_color = self._render_textures[-1].texture is not None
+
+        # Sample
+        encoder = self._device.create_command_encoder()
+        self._copy_pixel(encoder, self._depth_texture, float_pos, 0)
+        if can_sample_color:
+            self._copy_pixel(encoder, self._render_textures[-1], float_pos, 4)
+        queue = self._device.default_queue
+        queue.submit([encoder.finish()])
+
+        # Collect data from the buffer
+        data = self._pixel_info_buffer.read_data()
+        raw_depth = data[0:4].cast("f")[0]
+
+        # Calculate world coordinates using the camera
+        # todo: how does this hold up when a renderer is used with multiple cameras?
+        camera = self._last_camera
+        pos = Vector3(2 * float_pos[0] - 1, 2 * float_pos[1] - 1, raw_depth)
+        pos.apply_matrix4(camera.projection_matrix_inverse)
+        pos.apply_matrix4(camera.matrix_world)
+
+        return {
+            "rgba": tuple(data[4:8].cast("B")) if can_sample_color else (0, 0, 0, 0),
+            "depth_value": raw_depth,
+            "position": tuple(pos.to_array()),
+            "distance_to_camera": pos.distance_to(camera.position),
+        }
+
+    def _copy_pixel(self, encoder, render_texture, float_pos, buf_offset):
+
+        # Map position to the texture index
+        w, h, d = render_texture.size
+        x = max(0, min(w - 1, int(float_pos[0] * w)))
+        y = max(0, min(h - 1, int(float_pos[1] * h)))
+
+        bytes_per_pixel = 4  # todo: let's derive this from the format?
+
+        encoder.copy_texture_to_buffer(
+            {
+                "texture": render_texture.texture,
+                "mip_level": 0,
+                "origin": (x, y, 0),
+            },
+            {
+                "buffer": self._pixel_info_buffer,
+                "offset": buf_offset,
+                "bytes_per_row": bytes_per_pixel,
+                "rows_per_image": 1,
+            },
+            copy_size=(1, 1, 1),
+        )
+
+    def snapshot(self, step_index=-1):
+        """Create a snapshot of the currently rendered image."""
+
```

Comment:
There might be a better name for this method. Ideas?

Reply:
screenshot? Though ironic if there's no screen

Reply:
Haha, yeah I want to avoid the term "screen". When there is a canvas, you can also create a snapshot, but it's not exactly the same as what you see on screen; it will be the (perhaps higher resolution) version from what is rendered to screen with the ssaa pass.

Reply:
Maybe `save()`? But that is so generic...

Reply:
I think snapshot is fine 👍